llvm-external-projects/iree-dialects/lib/Dialect/VectorExt/IR/VectorExtAttrs.cpp

```diff
@@ -196,6 +196,13 @@ uint64_t LayoutAttr::getShuffleOffset(int64_t reductionDim) {
   return offset;
 }
 
+static SmallVector<int64_t> getPermutedVector(ArrayRef<int64_t> vec,
+                                              ArrayRef<int64_t> permutation) {
+  SmallVector<int64_t> permutedVec(vec);
+  applyPermutationToVector(permutedVec, permutation);
```

Comment:
you can use
```
llvm::to_vector(llvm::map_range(permutation, [&](int64_t i) {
  return vec[i];
})));
```
and maybe just inline this helper to avoid the extra smallvectors.

Reply:
Internally, this calls the same thing, so not really different.  Also, I think there is only 1 copy, which cannot be avoided. I'm avoiding inlining it because I'm going to use it for NestedLayoutAttr::permute also (in a subsequent oatch).

Reply:
Actually you can just use the version on arrayrefs directly https://github.com/llvm/llvm-project/blob/73159a994abcbf82881ee15b0df5cf13c9671f31/mlir/include/mlir/Dialect/Utils/IndexingUtils.h#L201

Reply:
Interesting. I was trying, but it wasn't working. Let me try again, thanks!

Reply:
Ah i was using applyPermutationToVector instead of applyPermutation